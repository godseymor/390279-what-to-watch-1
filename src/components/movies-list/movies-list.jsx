import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import MovieCard from '../movie-card/movie-card.jsx';

class MoviesList extends PureComponent {

  render() {
    const {activeItem, films, onChange, onTitleClick} = this.props;
    return (
      <div className="catalog__movies-list">
        {films.map((movie, i) =>
          <MovieCard
            movie={movie}
            key={i}
            onTitleClick={onTitleClick}
            isPlaying={movie.movieId === activeItem}
            onChange={onChange}
          />)}
      </div>
    );
  }
}


MoviesList.propTypes = {
  activeItem: PropTypes.string,
  onChange: PropTypes.func,
  films: PropTypes.array.isRequired,
  onTitleClick: PropTypes.func
};

const mapStateToProps = (state, ownProps) =>
  Object.assign({}, ownProps, {
    films: state.films
  });

export {MoviesList};

export default connect(mapStateToProps)(MoviesList);
