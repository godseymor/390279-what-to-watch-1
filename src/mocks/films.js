export default [
  {
    movieId: `1`,
    movieTitle: `Fantastic Beasts: The Crimes of Grindelwald`,
    movieImageSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    moviePreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    movieGenre: `comedy`
  },
  {
    movieId: `2`,
    movieTitle: `Bohemian Rhapsody`,
    movieImageSrc: `img/bohemian-rhapsody.jpg`,
    moviePreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    movieGenre: `crime`
  },
  {
    movieId: `3`,
    movieTitle: `Macbeth`,
    movieImageSrc: `img/macbeth.jpg`,
    moviePreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    movieGenre: `documentary`
  },
  {
    movieId: `4`,
    movieTitle: `Aviator`,
    movieImageSrc: `img/aviator.jpg`,
    moviePreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    movieGenre: `drama`
  },
  {
    movieId: `5`,
    movieTitle: `We need to talk about Kevin`,
    movieImageSrc: `img/we-need-to-talk-about-kevin.jpg`,
    moviePreview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    movieGenre: `horror`
  }
];
