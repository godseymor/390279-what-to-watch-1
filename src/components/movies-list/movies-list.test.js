import React from 'react';
import renderer from 'react-test-renderer';
import {MoviesList} from './movies-list';
import films from '../../mocks/films';

it(`moviesList renders correctly`, () => {

  const tree = renderer
    .create(<MoviesList
      films={films}
      onTitleClick={jest.fn()}
      onHover={jest.fn()}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
