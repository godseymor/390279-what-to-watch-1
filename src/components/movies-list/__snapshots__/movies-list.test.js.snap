// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moviesList renders correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        muted={true}
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        muted={true}
        poster="img/bohemian-rhapsody.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        muted={true}
        poster="img/macbeth.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Macbeth
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        muted={true}
        poster="img/aviator.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        height={175}
        muted={true}
        poster="img/we-need-to-talk-about-kevin.jpg"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
      onClick={[MockFunction]}
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>
</div>
`;
